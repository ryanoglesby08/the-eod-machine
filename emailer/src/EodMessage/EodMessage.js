import React from 'react'

import gql from 'graphql-tag'
import { Query } from 'react-apollo'

export const GET_EOD = gql`
  {
    eod {
      entries {
        category
        content
      }
    }
  }
`

const EodEmail = () => (
  <Query query={GET_EOD}>
    {({ data: { eod } }) => {
      const savedEntries = eod ? eod.entries : []

      return (
        <div>
          {savedEntries.map(({ category, content }) => (
            <div key={content}>
              {category}: {content}
            </div>
          ))}
        </div>
      )
    }}
  </Query>
)

export default EodEmail
