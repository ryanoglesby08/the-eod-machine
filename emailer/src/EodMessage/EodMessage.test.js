import React from 'react'

import { MockedProvider } from 'react-apollo/lib/test-utils'
import { render, waitForElement } from 'react-testing-library'

import EodMessage, { GET_EOD } from './EodMessage'

it('is an EOD message', async () => {
  const mocks = [
    {
      request: {
        query: GET_EOD,
      },
      result: {
        data: {
          eod: {
            entries: [{ category: 'Test category', content: 'test content' }],
          },
        },
      },
    },
  ]

  const { container } = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <EodMessage />
    </MockedProvider>
  )

  await waitForElement()

  expect(container).toMatchSnapshot()
})
