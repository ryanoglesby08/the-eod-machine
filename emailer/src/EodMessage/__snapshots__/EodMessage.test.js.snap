// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is an EOD message 1`] = `
<div>
  <div>
    <div>
      Test category
      : 
      test content
    </div>
  </div>
</div>
`;
